raiden/transfer/mediated_transfer/mediator.py

```diff
@@ -243,7 +244,12 @@ def get_lock_amount_after_fees(
     # we only have the amount including fee_out, so we use that as an
     # approximation.
     fee_out = _fee_for_channel(payee_channel, PaymentAmount(lock.amount - fee_in))
-    return PaymentWithFeeAmount(lock.amount - fee_in - fee_out)
+    amount_after_fees = PaymentWithFeeAmount(lock.amount - fee_in - fee_out)
+    if amount_after_fees <= 0:
+        raise TransferAmountSmallerThanFees(
```

Comment:
Please don't raise exceptions from the state machine, this is not caught and will crash the client.

The best approach here would be to return a `Optional[PaymentWithFeeAmount]`